crates/deltalake-core/src/kernel/actions/schemas.rs

```diff
@@ -9,236 +9,241 @@ lazy_static! {
     // https://github.com/delta-io/delta/blob/master/PROTOCOL.md#change-metadata
     static ref METADATA_FIELD: StructField = StructField::new(
         "metaData",
-        DataType::Struct(Box::new(StructType::new(vec![
```

Comment:
These changes look like they include nullable column changes, which will require a little finer review than I have time for today before the merge. I don't see any big problems, but I do see some differences that I need to noodle on :thinking:

Reply:
they do - this is mainly due to the fact that these schemas represent what we read from checkpoint / log files, and as such the nullability of the column in the whole record-batch. Thus here almost everything is nullable.

The actual nullability of the data structures is only enforced once we parse that data. there we then error for missing required fields.